tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,635 @@
+    assert check_instance(data) == expected
+
+
+def test_nested_dict():
+    data = {"x": {"y": 2}}
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_numpy():
+    data = {"x": np.array([])}
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_numpy_1d():
+    data = {"x": np.array([2, 3])}
+    expected = pd.DataFrame(np.array([2, 3]), columns=['x'])
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_numpy_2d():
+    data = {"x": np.array([[2, 3]])}
+    expected = pd.DataFrame(np.array([[2, 3]])).add_prefix("x_")
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_numpy_gt_2d():
+    data = {"x": np.array([[[2, 3]]])}
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_series_empty():
+    """Test that values in key value pair should not be empty ... for Series"""
+    data = {"x": pd.Series([], dtype='int')}
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_series_not_multiIndex_no_name():
+    """Test for single index series"""
+    data = {"x": pd.Series([2, 3])}
+    expected = pd.DataFrame([2, 3], columns=["x"])
+    assert_frame_equal(check_instance(data)[0][0], expected)
+
+
+def test_series_not_multiIndex_with_name():
+    """Test for single index series with name"""
+    data = {"x": pd.Series([2, 3], name="y")}
+    expected = pd.DataFrame([2, 3], columns=["x_y"])
+    assert_frame_equal(check_instance(data)[0][0], expected)
+
+
+def test_series_multiIndex():
+    """Test that multiIndexed series trigger error"""
+    data = {
+        "x": pd.Series([2, 3],
+                       index=pd.MultiIndex.from_arrays([[1, 2], [3, 4]]))
+    }
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_dataframe_empty():
+    """Test for empty dataframes"""
+    data = {"x": pd.DataFrame([])}
+    with pytest.raises(ValueError):
+        assert expand_grid(data)
+
+
+def test_dataframe_single_index():
+    """Test for single indexed dataframes"""
+    data = {"x": pd.DataFrame([[2, 3], [6, 7]])}
+    expected = pd.DataFrame([[2, 3], [6, 7]]).add_prefix("x_")
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_dataframe_multiIndex_index():
+    """Test for multiIndex dataframe"""
+    data = {
+        "x":
+        pd.DataFrame([[2, 3], [6, 7]],
+                     index=pd.MultiIndex.from_arrays([['a', 'b'], ['y', 'z']]))
+    }
+
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_dataframe_multiIndex_column():
+    data = {
+        "x":
+        pd.DataFrame([[2, 3], [6, 7]],
+                     columns=pd.MultiIndex.from_arrays([['m', 'n'], ['p',
+                                                                     'q']]))
+    }
+
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_dataframe_multiIndex_index_column():
```

Comment:
```suggestion
def test_dataframe_multi_index_index_column():
```

Could you add a docstring to tells us what the test is intended to test?